src/middlewared/middlewared/plugins/ports/ports.py

```
@@ -0,0 +1,36 @@
+from middlewared.service import Service, ValidationErrors
+
+
+class PortService(Service):
+
+    DELEGATES = {}
+    SYSTEM_USED_PORTS = [
+        {'title': 'System', 'ports': [67, 123, 3702, 5353, 6000], 'namespace': 'system'},
+    ]
+
+    class Config:
+        private = True
+
+    async def register_attachment_delegate(self, delegate):
+        self.DELEGATES[delegate.namespace] = delegate
```

Should we raise an error if a delegate is already registered (key exists)?